lib/integrations/awsoidc/idp_iam_config_test.go

```diff
@@ -558,6 +559,9 @@ func (m *mockIdPIAMConfigClient) TagRole(ctx context.Context, params *iam.TagRol
 
 func TestNewIdPIAMConfigureClient(t *testing.T) {
 	t.Run("no aws_region env var, returns an error", func(t *testing.T) {
+		t.Setenv("AWS_REGION", "")
+		os.Unsetenv("AWS_REGION")
```

Comment:
What's the point of the `Unsetenv` here?

Reply:
The test requires that the `AWS_REGION` env var is not set (not just empty).

However, we don't want to change anything on the underlying system, so we need to
- store the existing value for the env var
- unset it
- restore the value into the env var

Calling `t.Setenv` does the 1st and 3rd steps.
It changes the env var and adds a cleanup function that will restore its value.
And the 2nd step is achieved by the `os.Unsetenv`

Reply:
What cares about the difference between unset and empty string? Is that Teleport code that cares or AWS code?

Reply:
AWS code